extensions/resource-deployment/src/interfaces.ts

```diff
@@ -153,18 +156,32 @@ export interface DialogInfoBase {
 export interface DialogTabInfo extends PageInfoBase {
 }
 
-export interface PageInfoBase extends SharedFieldAttributes {
+export interface PageInfoBase extends InheritableAttributes {
 	title: string;
 	isSummaryPage?: boolean;
 	sections: SectionInfo[];
 }
 
-export interface SharedFieldAttributes {
+export interface TextCSSStyles {
+	fontStyle?: FontStyle | undefined;
+	fontWeight?: FontWeight | undefined;
+	color?: string;
+	[key: string]: string | undefined;
+}
+
+export type ComponentCSSStyles = {
+	[key: string]: string;
+};
+
+export interface InheritableAttributes {
 	labelWidth?: string;
 	inputWidth?: string;
 	labelPosition?: LabelPosition; // Default value is top
+	rowWidth?: string;
```

Comment:
the term row here might not be appropriate, fieldWidth?

Reply:
I have renamed to fieldWidth for this check-in. However, labelWidth, inputWidth, fieldWidth is all a lot of widths. I will factor out width, height, layout etc as separate properties that can be applied to label, field, input component, row etc. But I will do that in a separate check-in so that I can keep various aspects of functionality independent.